lib/domain/libCircuit/langs/circom/CircomModel.h

```diff
@@ -0,0 +1,134 @@
+//
+// Created by Daniel Dominguez on 17/11/2022.
+//
+
+#ifndef VANGUARD_CIRCOMMODEL_H
+#define VANGUARD_CIRCOMMODEL_H
+
+#include "domain/libCircuit/CircuitModel.h"
+#include "CircomUtils.h"
+
+namespace vanguard {
+
+    template <typename Domain>
+    class CircomModel : public CircuitModel<Domain> {
+    public:
+        CircomModel(const llvm::Module &module, typename Domain::Factory &factory) : module(module), factory(factory) {};
+
+        typename Domain::Type *getComponentType(std::string name) override {
+            std::string structName = vanguard::componentNameToStructName(name);
+            for (auto strct : module.getIdentifiedStructTypes()) {
+                if (strct->hasName() && strct->getName().str() == structName) {
+                    return factory.createType(strct);
+                }
+            }
+            return nullptr;
+        }
+
+        class SignalIndexSearcher : public InstructionClassVisitor<Domain> {
+        public:
+            SignalIndexSearcher(int index, typename Domain::Factory &factory, typename Domain::Type *componentType)
+                : index(index), found(false), value(nullptr), factory(factory), componentType(componentType) {};
+
+//            virtual void visit(const BranchIns<Domain> &v) {};
+//            virtual void visit(const ReturnIns<Domain> &v) {};
+//            virtual void visit(const ErrorIns<Domain> &v) {};
+//            virtual void visit(const BinaryOpIns<Domain> &v) {};
+//            virtual void visit(const UnaryOpIns<Domain> &v) {};
+//            virtual void visit(const CallIns<Domain> &v) {};
+//            virtual void visit(const CastIns<Domain> &v) {};
+//            virtual void visit(const TernaryIns<Domain> &v) {};
+//            virtual void visit(const AssignIns<Domain> &v) {};
+            virtual void visit(const UnknownIns<Domain> &v) {
+                int probe = 0;
+                if (!llvm::isa<llvm::GetElementPtrInst>(v.unwrap())) return;
+                auto llvmInst = llvm::dyn_cast<llvm::GetElementPtrInst>(v.unwrap());
```

Comment:
nit: `llvm::dyn_cast` returns null if it is not the given type, so this could be more cleanly expressed as:
```suggestion
                auto llvmInst = llvm::dyn_cast<llvm::GetElementPtrInst>(v.unwrap());
                if (!llvmInst) return;
```